### ISSUE TITLE
gccrs can't compile let statment correctly

### ISSUE THREAD

**Issue by zhanghe9702** [Rust-GCC/gccrs#2915]
I tried this code:

```rust
fn get_myname() -> Option<String> {
   return Some("myname".to_string());
}

#[allow(unused_variables)]
fn main() {
    let Some(myname) = get_myname()  else {
        panic!();
    };
}
and compile using gccrs with follow command line:
./gccrs-install/bin/gccrs -frust-edition=2018 -frust-incomplete-and-experimental-compiler-do-not-use -c -o test.s  ./test.rs

```

I expected to see this happen: *compile the file correctly*

Instead, this happened: 
```
./test.rs:7:38: error: expecting ';' but 'else' found
    7 |     let Some(myname) = get_myname()  else {
      |                                      ^~~~
./test.rs:7:5: error: failed to parse statement or expression in block expression
    7 |     let Some(myname) = get_myname()  else {
      |     ^~~
./test.rs:7:38: error: unrecognised token 'else' for start of item
    7 |     let Some(myname) = get_myname()  else {
      |                                      ^~~~
./test.rs:7:38: error: failed to parse item in crate
```
by the way  , rustc could generate asm and with no problem 

### Meta

- What version of Rust GCC were you using, git sha if possible.
75dc00584e0e7a73ddcbef358abae24762b27bcd

**Comment by zhanghe9702** [comment]
as rust refenrence:
LetStatement :
   [OuterAttribute](https://doc.rust-lang.org/reference/attributes.html)* let [PatternNoTopAlt](https://doc.rust-lang.org/reference/patterns.html) ( : [Type](https://doc.rust-lang.org/reference/types.html) )? (= [Expression](https://doc.rust-lang.org/reference/expressions.html) [†](https://doc.rust-lang.org/reference/statements.html#let-else-restriction) ( else [BlockExpression](https://doc.rust-lang.org/reference/expressions/block-expr.html)) ? ) ? ;

https://github.com/Rust-GCC/gccrs/blob/53d8e04c0b4a4834658397dc61352cf8928c6b51/gcc/rust/parse/rust-parse-impl.h#L6248-L6265

is this the problem? we han't consume `else` part in parser ?

**Comment by CohenArthur** [comment]
this is a complex problem - `let-else` are only stable since Rust 1.65, while we target Rust 1.49. We'll need to handle them properly eventually but there's not a lot of infrastructure for them at the moment. We do need to modify the parser to accept them, so that would be a good first task for fixing that issue :)

**Comment by zhanghe9702** [comment]
for let-else ast node lowering to tree node , looks like should refactor HIR::LetStmt process ?

**Comment by CohenArthur** [comment]
yes, we'll need to refactor both the `HIR::LetStmt` and `AST::LetStmt` as they both can't handle `else` expressions
